### PR TITLE
chore(golang): update to 1.18, fixes to do that, update deps, particularly nomad

### PR DESCRIPTION
**Pull request by mgeggie**
This was able to successfully build scalad and seems to be built against a 2023 version of Nomad API.

We could try posting this container to our zumper Docker registry to see if it'll run in test.

I'm not sure if we should post this to upstream trivago/scalad, since it seems like they've largely abandoned the project.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,17 @@
+module github.com/trivago/scalad
```
Comment (mgeggie) [inline go.mod:1]: This file is entirely created by running `go mod tidy`

File: go.sum [hunk 2]
```diff
@@ -0,0 +1,794 @@
+cloud.google.com/go v0.26.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
```
Comment (mgeggie) [inline go.sum:1]: This file is also entirely created by `go mod tidy`

**Review by jvineet — APPROVED** [review]

**Review by kyledicesare — APPROVED** [review]
LGTM

**Review by tetsujiono — APPROVED** [review]
lgtm

**Pull request merged**